cfme/tests/cloud_infra_common/test_retirement.py

```diff
@@ -27,78 +28,83 @@
 
 
 @pytest.yield_fixture(scope="function")
-def vm(small_template, provider):
-    vm_obj = VM.factory(random_vm_name('retire'), provider, template_name=small_template)
-    vm_obj.create_on_provider(find_in_cfme=True, allow_skip="default")
-    yield vm_obj
+def test_vm(small_template, provider):
+    test_vm = VM.factory(random_vm_name('retire'), provider, template_name=small_template)
+    test_vm.create_on_provider(find_in_cfme=True, allow_skip="default")
+    yield test_vm
 
     try:
-        if provider.mgmt.does_vm_exist(vm_obj.name):
-            provider.mgmt.delete_vm(vm_obj.name)
+        if provider.mgmt.does_vm_exist(test_vm.name):
+            provider.mgmt.delete_vm(test_vm.name)
     except Exception:
-        logger.warning('Failed to delete vm from provider: {}'.format(vm_obj.name))
+        logger.warning('Failed to delete vm from provider: {}'.format(test_vm.name))
 
 
-@pytest.yield_fixture(scope="function")
-def existing_vm(provider):
-    """ Fixture will be using for set\unset retirement date for existing vm instead of
-    creation a new one
-    """
-    all_vms = provider.mgmt.list_vm()
-    need_to_create_vm = True
-    for virtual_machine in all_vms:
-        if provider.mgmt.is_vm_running(virtual_machine):
-            vm_obj = VM.factory(virtual_machine, provider)
-            need_to_create_vm = False
-            break
-    if need_to_create_vm:
-        machine_name = random_vm_name('retire')
-        vm_obj = VM.factory(machine_name, provider)
-        vm_obj.create_on_provider(find_in_cfme=True, allow_skip="default")
-
-    yield vm_obj
+def verify_retirement_state(test_vm):
+    # wait for the info block showing a date as retired date
+    # Use lambda for is_retired since its a property
+    assert wait_for(lambda: test_vm.is_retired, delay=5, num_sec=10 * 60, fail_func=tb.refresh,
+             message="Wait for VM '{}' to enter retired state".format(test_vm.name))
 
-    try:
-        if need_to_create_vm and provider.mgmt.does_vm_exist(vm_obj.name):
-            provider.mgmt.delete_vm(vm_obj.name)
-    except Exception:
-        logger.warning('Failed to delete vm from provider: {}'.format(vm_obj.name))
+    retire_time = format_retirement_date_now()
 
+    # TODO: remove dependency on SummaryMixin and use widgetastic when available
+    assert test_vm.summary.power_management.power_state.text_value in ['off', 'suspended',
+                                                                       'unknown']
+    return retire_time
```

Comment:
This is kind of screwy but I use it to try and get an accurate timestamp for retirement. There are definitely other and better ways to approach this. It's a nuance that I don't quite feel like really killing right now.